### DISCUSSION TITLE
Valetudo Local Map Extractor

### DISCUSSION THREAD

**Discussion by sca075** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#428] [Q&A]
Dear @PiotrMachowski 
You made a really great job with this custom component and honestly instead to re-invent a component I would like to make the camera that I currently work on to be 100% compatible with your integration. 
The idea.. come with full local control concept. So.. Valetudo (that honestly has not a really nice way to offer the data to HA)... as an API from witch it is possible to get the map data.
My robot is a Xaiomi Robrock V1. Therefore it uses just zone cleaning. 
There are out other way to get the map out of it.. but pulling the json gave me the result you can see [below.]

 
![Screenshot 2023-04-23 at 15 16 32](https://user-images.githubusercontent.com/82227818/233842453-c0417481-3217-46c9-91b9-de680546e92b.png)

I´m using actually one old code of the first camera you made.. with some changes of course. I use PIL draw and so far the complete download of the Jason and image creation is down to 5sec . I would just need to know what data the camera should send back to the card to establish it as a "calibrated" one.. I mean.. cropped the map there is some difference in the real points as per valetudo standard use 5120 * 5120 maps. It was my idea to send back the offset created instead of sending the full image back that slow down everything. I would share gladly the code of course, as soon I fix some minor bug / stability issues.. Waiting for your kind replay, thanks in advance.

**Comment by PiotrMachowski** [comment]
Hi @sca075!

Thanks for you input, you can check out this pull request:
https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/pull/408
